ui/raidboss/data/06-ew/raid/p4s.ts

```diff
@@ -0,0 +1,704 @@
+      condition: (data) => data.decOffset === undefined,
+      // Unconditionally set the first headmarker here so that future triggers are conditional.
+      run: (data, matches) => getHeadmarkerId(data, matches),
+    },
+    {
+      id: 'P4S Decollation',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6A09', source: 'Hesperos', capture: false }),
+      netRegexDe: NetRegexes.startsUsing({ id: '6A09', source: 'Hesperos', capture: false }),
+      netRegexFr: NetRegexes.startsUsing({ id: '6A09', source: 'Hespéros', capture: false }),
+      netRegexJa: NetRegexes.startsUsing({ id: '6A09', source: 'ヘスペロス', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'P4S Bloodrake',
+      // AoE hits tethered players in first one, the non-tethered in second
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6A40', source: 'Hesperos', capture: false }),
+      netRegexDe: NetRegexes.startsUsing({ id: '6A40', source: 'Hesperos', capture: false }),
+      netRegexFr: NetRegexes.startsUsing({ id: '6A40', source: 'Hespéros', capture: false }),
+      netRegexJa: NetRegexes.startsUsing({ id: '6A40', source: 'ヘスペロス', capture: false }),
+      response: Responses.aoe(),
+    },
+    {
+      id: 'P4S Elegant Evisceration',
+      // This one does an aoe around the tank
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '6A08', source: 'Hesperos' }),
+      netRegexDe: NetRegexes.startsUsing({ id: '6A08', source: 'Hesperos' }),
+      netRegexFr: NetRegexes.startsUsing({ id: '6A08', source: 'Hespéros' }),
+      netRegexJa: NetRegexes.startsUsing({ id: '6A08', source: 'ヘスペロス' }),
+      response: Responses.tankBusterSwap('alert'),
+    },
+    {
+      id: 'P4S Levinstrike Pinax',
+      // Strong proximity Aoe
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '69D7', source: 'Hesperos', capture: false }),
+      netRegexDe: NetRegexes.startsUsing({ id: '69D7', source: 'Hesperos', capture: false }),
+      netRegexFr: NetRegexes.startsUsing({ id: '69D7', source: 'Hespéros', capture: false }),
+      netRegexJa: NetRegexes.startsUsing({ id: '69D7', source: 'ヘスペロス', capture: false }),
+      alertText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Go to Corner',
+          de: 'In eine Ecke gehen',
+          fr: 'Allez dans un coin',
+          cn: '去角落',
+          ko: '구석으로',
+        },
+      },
+    },
+    {
+      id: 'P4S Well Pinax',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '69D6', source: 'Hesperos', capture: false }),
+      netRegexDe: NetRegexes.startsUsing({ id: '69D6', source: 'Hesperos', capture: false }),
+      netRegexFr: NetRegexes.startsUsing({ id: '69D6', source: 'Hespéros', capture: false }),
+      netRegexJa: NetRegexes.startsUsing({ id: '69D6', source: 'ヘスペロス', capture: false }),
+      delaySeconds: 5,
```

Comment:
```
[21:28:20.488] StartsCasting 14:40019333:Hesperos:69D6:Well Pinax:40019333:Hesperos:8.70:90.00:90.00:0.00:0.00
[21:28:23.605] StartsCasting 14:40019303:Hesperos:69FE:Southerly Shift:40019303:Hesperos:7.70:99.99:99.99:0.00:-1.30
[21:28:29.477] ActionEffect 15:40019333:Hesperos:69D6:Well Pinax:E0000000::0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:0:::::::::::44:44:0:10000:::90.00:90.00:0.00:0.00:000163D6:0:0
[21:28:31.571] ActionEffect 15:40019303:Hesperos:69FE:Southerly Shift:40019303:Hesperos:1B:69FE8000:0:0:0:0:0:0:0:0:0:0:0:0:0:0:35230907:41780192:10000:10000:::99.99:99.99:0.00:3.14:35230907:41780192:10000:10000:::99.99:99.99:0.00:3.14:000163E2:0:1
```

There is roughly 11 seconds between Well Pinax cast start and Southerly Shift, so maybe delay 6?

Similarly, Southerly Shift is an 8s cast, so maybe delay 3?

Reply:
Hmm.  Sorry, I don't think the generalizations you wrote here work for this fight.  I think it's important that knockback calls cover both knockbacks if possible.

Well Pinax is a 9 second cast (and castTime=8.7 here, as there's always 0.3 missing for whatever reason).  If you give a 5 second warning, then:
* t=0, Well Pinax starts casting
* t=3.1 Southerly Shift starts casting
* t=3.7 "Knockback" trigger 5 seconds ahead (8.7 - 5 = 3.7)
* t=5.1 (earliest a knockback call could go and still cover Southerly Shift, probably more like 5.6 is a better compromise?)
* t=5.8 "South Knockback" trigger (7.7 - 5 = 2.7, this is fine)
* t=9 Well Pinax goes off
* t=9.7 (Well Pinax knockback wears off, if used at 3.7)
* t =11.1 Southerly Shift goes off
* t=11.8 (Southerly Shift knockback wears off, if used at 5.8)

I said before delaySeconds 6 and 3, but maybe a compromise is Well Pinax 5.5 and Southerly Shift 2.5?

Reply:
Yea, might be one where it's going to take some experience to get the timing down. I have updated it to 5.5 and 2.5.